app/domain/authentication/handler/authentication_handler.rb

```diff
@@ -130,7 +130,14 @@ def validate_client_ip(client_ip_address, conjur_role)
       end
 
       def fetch_conjur_role(authenticator, identity)
-        return @role_repository_class.from_username(authenticator.account, identity)
+        annotation = Annotation.fetch_annotation(
```

Comment:
Instead of including the `Annotation` dependency here, I'd recommend injecting it into the `Authentication::Handler::AuthenticationHandler` initialization method.  

This will allow you to write tests for this handler without loading anything into the database.